cluster/lock.go

```diff
@@ -203,6 +211,45 @@ func (l Lock) VerifySignatures() error {
 	return nil
 }
 
+// verifyBuilderRegistrations returns an error if populated builder registrations from json are invalid.
+func (l Lock) verifyBuilderRegistrations() error {
+	for i, val := range l.Validators {
+		// Check if the current cluster state supports pre-generate validator registrations.
+		if len(val.BuilderRegistration.Signature) == 0 ||
+			len(val.BuilderRegistration.Message.FeeRecipient) == 0 ||
+			len(val.BuilderRegistration.Message.PubKey) == 0 {
+			continue
+		}
+
+		regMsg, err := registration.NewMessage(eth2p0.BLSPubKey(val.PubKey), l.FeeRecipientAddresses()[i], uint64(val.BuilderRegistration.Message.GasLimit), val.BuilderRegistration.Message.Timestamp)
```

Comment:
nit: can store the `fee-recipient-addresses` instead of calling `l.FeeRecipientAddresses()` in each iteration